bittide-instances/src/Bittide/Instances/Hitl/FullMeshSwCc.hs

```diff
@@ -188,20 +200,71 @@ fullMeshHwTest refClk sysClk IlaControl{syncRst = rst, ..} rxNs rxPs miso =
 
   transceivers =
     transceiverPrbsN
-      @GthTx @GthRx @Ext200 @Basic125 @GthTx @GthRx
+      @GthTx @GthRx @Ext200 @Basic125 @GthTxS @GthRxS @LinkCount
       Transceiver.defConfig
       Transceiver.Inputs
         { clock = sysClk
         , reset = gthAllReset
         , refClock = refClk
-        , channelNames
-        , clockPaths
+        , channelNames = takeI channelNames
+        , clockPaths = takeI clockPaths
         , rxNs
         , rxPs
-        , txDatas = repeat (pure 0)
-        , txReadys = repeat (pure False)
+        , txDatas = txCounters
+        , txReadys = txAllStables
         , rxReadys = repeat (pure True)
         }
+  txAllStables = zipWith (xpmCdcSingle sysClk) transceivers.txClocks (repeat allStable1)
+  allStable1 = sticky sysClk syncRst allStable0
+  txResets2 = zipWith orReset
+    transceivers.txResets
+    (map unsafeFromActiveLow txAllStables)
+
+  txCounters = zipWith3 ugnCounter transceivers.txClocks txResets2 transceivers.txSamplings
+  ugnCounter txClk txRst txSampl = result
+   where
+    result = register txClk txRst (toEnable txSampl) (0xaabbccddeeff1234 :: BitVector 64) (result + 1)
+
+  -- rxFifos :: Vec LinkCount (_, _, _, _, _Signal GthRx (Maybe (BitVector 64)))
+  rxFifos = zipWith4 go transceivers.txClocks transceivers.rxClocks
+                   txResets2
+                   transceivers.rxDatas
+   where
+    go txClk rxClk txRst rxData = resettableXilinxElasticBuffer @FifoSize @_ @_ @(Maybe (BitVector 64)) txClk rxClk txRst rxData
+
+  (fillLvls,fifoUnderflowsTx,fifoOverflowsTx,_ebMode,rxCntrs) = unzip5 rxFifos
+
+  fifoOverflowsFree :: Vec LinkCount (Signal Basic125 Overflow)
+  fifoOverflowsFree = zipWith (flip xpmCdcSingle sysClk) transceivers.txClocks fifoOverflowsTx
+  fifoUnderflowsFree :: Vec LinkCount (Signal Basic125 Underflow)
+  fifoUnderflowsFree = zipWith (flip xpmCdcSingle sysClk) transceivers.txClocks fifoUnderflowsTx
+
+
+  ugns :: Vec LinkCount (Signal GthTx (BitVector 64))
+  ugns = zipWith (-) txCounters
+              (map (fmap (fromMaybe 0x1122334411223344)) rxCntrs)
```

Comment:
Please add a comment explaining the `0x1122334411223344`

Reply:
For a properly working system it's meaningless, it could be anything and initally it used to be `0`.
But at some point during debugging I wanted a easy way of seeing if the transceivers were outputting `Nothing` or `Just 0`, both of which resulted in `ugn = txCounter - 0`.

It goes together with the starting value of txCounter: `0xaabbccddeeff1234`.
Same story, could be anything.

`0xaabbccddeeff1234 - 0x1122334411223344 = 0x99999999dddcdef0`

If you ever the ugn being a constant `0x99999999dddcdef0` then you know the your counter isn't running and you're receiving `Nothing`, (or you won the 1 in 2^64 lottery).
If you seeing ``0x99999999..........`` and it's counting up, then your receiving `Nothing`, but your counter is running.

Reply:
And you can still see this in the ILA dump now.
Before the links are up (`probe_Linkups == 0`) all ugns are `0x99999999dddcdef0`.

Reply:
Yeah makes sense. Could you add these considerations as a comment in the code?